tools/integration_tests/implicitdir/file_operations_test.go

```diff
@@ -73,3 +74,45 @@ func TestFileAttributes(t *testing.T) {
 		t.Errorf("File size is not 14 bytes, found size: %d bytes", fStat.Size())
 	}
 }
+
+func TestCopyFile(t *testing.T) {
+	fileName := createTempFile()
+	err := clearKernelCache()
+	if err != nil {
+		t.Errorf("Clear Kernel Cache: %v", err)
+	}
+	content, err := os.ReadFile(fileName)
+	if err != nil {
+		t.Errorf("Read: %v", err)
+	}
+	newFileName := fileName + "Copy"
+	if _, err := os.Stat(newFileName); err == nil {
+		t.Errorf("Copied file %s already present", newFileName)
+	}
+
+	// File copying with io.Copy() utility.
+	source, err := os.Open(fileName)
+	if err != nil {
+		t.Errorf("File %s opening error: %v", fileName, err)
+	}
+	defer source.Close()
+	destination, err := os.Create(newFileName)
+	if err != nil {
+		t.Errorf("Copied file creation error: %v", err)
+	}
+	defer destination.Close()
+	_, err = io.Copy(destination, source)
+	if err != nil {
+		t.Errorf("Error in file copying: %v", err)
+	}
+
+	// Checks on old and new file.
+	if _, err := os.Stat(fileName); os.IsExist(err) {
+		t.Errorf("Old file %s not found", fileName)
+	}
+	if _, err := os.Stat(newFileName); os.IsNotExist(err) {
+		t.Errorf("Copied file %s not found", newFileName)
+	}
```

Comment:
i would suggest removing stat checks, since if you are comparing the contents, you are implicitly checking if the file is present.